torch/_inductor/codegen/cpp.py

```diff
@@ -1236,6 +1236,15 @@ def index_to_str(self, index: sympy.Expr) -> str:
         """
         return cexpr(self.rename_indexing(index))
 
+    def get_ranges(self):
+        ranges = {}
+        for idx, upper in zip(self.itervars, self.ranges):
```

Comment:
Please avoid repeating code. This function can be lifted to common.py if you implement `var_ranges()` in cpp.py

Reply:
Also, we should probably cache this function on self. We have a decorator to do that